### PR TITLE
Added possibility to check if a directory is empty

### PR DESCRIPTION
**Pull request by ekkinox**
- Added possibility to check if a directory is empty via FileSystemComponent
- Added related tests
- Bump to version 8.2.0

### CODE REVIEW

File: common/configuration/class.FileSystemComponent.php [hunk 1]
```diff
@@ -1,325 +1,388 @@
...
+     *
+     * @access public
+     * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
+     * @param  string location
+     * @param  string expectedRights
+     * @param  boolean optional
+     * @throws common_configuration_MalformedRightsException
+     * @return mixed
+     */
+    public function __construct($location, $expectedRights, $optional = false, $recursive = false, $mustCheckIfEmpty = false)
```
[the remark this answers is not in the document]
Comment (ekkinox) [reply common/configuration/class.FileSystemComponent.php]: I wish :/

File: common/configuration/class.FileSystemComponent.php [hunk 2]
```diff
@@ -1,325 +1,388 @@
...
+                return new common_configuration_Report(
+                    common_configuration_Report::INVALID,
+                    "File system component '${name}' in '${location} is not executable.",
+                    $this
+                );
+            }
+
+            if ($this->getMustCheckIfEmpty()) {
+                if (!$this->isEmptyDirectory($location)) {
+                    return new common_configuration_Report(
```
[the remark this answers is not in the document]
Comment (ekkinox) [reply common/configuration/class.FileSystemComponent.php]: requires complete rework, but used by UX install, CLI install and seed install

File: common/configuration/class.FileSystemComponent.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (ekkinox) [reply common/configuration/class.FileSystemComponent.php]: requires complete rework, but used by UX install, CLI install and seed install.
I would avoid side effects

File: common/configuration/class.ComponentFactory.php [hunk 3]
```diff
@@ -1,470 +1,477 @@
...
+     * @param  string location
+     * @param  string expectedRights
+     * @param  boolean optional
+     * @param  boolean recursive
+     * @param  boolean mustCheckIfEmpty
+     * @throws common_configuration_MalformedRightsException
+     * @return common_configuration_FileSystemComponent
+     */
+    public static function buildFileSystemComponent(
+        $location,
```
[the remark this answers is not in the document]
Comment (ekkinox) [reply common/configuration/class.ComponentFactory.php]: requires complete rework, but used by UX install, CLI install and seed install
I would avoid side effects

**Comment by rkarolis** [comment]
Works on Windows well.

**Pull request merged**